File: solutions/_645_set_mismatch/solution.cpp
#include <set>
#include <vector>
using namespace std;

class Solution
{
public:
    vector<int> findErrorNums(vector<int> &nums)
    {
        set<int> s;
        int n = nums.size();
        int sum = 0;
        int duplicate = -1;
        for (int num : nums)
        {
            if (s.find(num) != s.end())
                duplicate = num;
            else
                s.insert(num);
            sum += num;
        }

        vector<int> result;
        int missing = (n * (n + 1) / 2) - (sum - duplicate);
        result.push_back(duplicate);
        result.push_back(missing);
        return result;
    }
};

/*
https://leetcode.com/problems/set-mismatch/
*/

File: solutions/_645_set_mismatch/solution.py
from typing import List


class Solution:
    def findErrorNums(self, nums: List[int]) -> List[int]:
        n = len(nums)
        s = set()

        for num in nums:
            if num in s:
                duplicate = num
            else:
                s.add(num)

        missing = n * (n + 1) // 2 - sum(nums) + duplicate
        return [duplicate, missing]


# https://leetcode.com/problems/set-mismatch/
